src/rmc-compiler/rustc_codegen_rmc/src/codegen/intrinsic.rs

```diff
@@ -29,6 +29,24 @@ impl<'tcx> GotocCtx<'tcx> {
         self.codegen_expr_to_place(p, e)
     }
 
+    /// Handles codegen for non returning intrinsics
+    /// Non returning intrinsics are not associated with a destination
+    pub fn codegen_nonret_intrinsic(
+        &mut self,
+        instance: Instance<'tcx>,
+        span: Option<Span>,
+    ) -> Stmt {
+        let intrinsic = self.symbol_name(instance);
+        let intrinsic = intrinsic.as_str();
+        let loc = self.codegen_span_option(span);
+        debug!("codegen_nonret_intrinsic:\n\tinstance {:?}\n\tspan {:?}", instance, span);
+        match intrinsic {
+            "abort" => Stmt::assert_false("reached intrinsic::abort", loc),
```

Comment:
I recommend using `GotocCtx::codegen_fatal_error()` here. The `Stmt::assert_false` will not abort the program execution path.